### ISSUE TITLE
Cross-platform build for Xenko projects

### ISSUE THREAD

**Issue by xen2** [stride3d/stride#360]
Idea is to be able to build Xenko projects (incl. asset compilation) on Linux/macOS.

Some important steps:
- [ ] Remove managed c++ code (esp. FBX/Assimp importers)
- [ ] Find/recompile/remove various native dependencies for Linux/macOS
- [ ] Convert projects to .NET Standard or .NET Core (to decide per project)

This is a necessary step before [cross-platform editor](https://github.com/xenko3d/xenko/issues/8)

**Comment by Kryptos-FR** [comment]
> Convert projects to .NET Core

.NET Standard should be enough for some of them, see my early attempts in https://github.com/Kryptos-FR/Avalonia.GameStudio

**Comment by xen2** [comment]
@Kryptos-FR yes sorry that's what I meant, editing

**Comment by dfkeenan** [comment]
That "cross-platform editor" link doesn't seem to go anywhere.

**Comment by SleepyMode** [comment]
> 
> 
> That "cross-platform editor" link doesn't seem to go anywhere.

It's a link to another issue.
#8

**Comment by xen2** [comment]
Yes my bad, it was broken before (I tried `[cross-platform editor](#8)` but it made a link on same page with `#8` in URL). Fixed it a few days ago.

**Comment by xen2** [comment]
More detailed steps:

**Runtime**
- [x] Update to latest `System` assemblies
- [x] Convert `Xenko.Games` and `Xenko.Input` to use .NET Core 3.0 WinForms & WPF, like on .NET Framework

** Compiler tools (.NET Core 3 on Windows)**
- [ ] Managed code (Assimp/FBX): if not converted to pure C/C#, need to do at least multi TFM build for .NET Core and .NET FW
- [ ] Review various use of EnvDTE (mostly used for debugger reattaching)
- [ ] Remove WCF code (maybe convert to [ServiceWire](https://github.com/tylerjensen/ServiceWire)?)

** Compiler tools (multiplatform) **
- [ ] Managed code (Assimp/FBX) needs to be converted to pure C/C#
- [ ] Rebuild various dependencies (Texture libraries, VHACD, msdfgen, etc.) for Linux/OSX

** Game Studio (.NET Core 3.0 on Windows) **
- [x] Update RoslynPad to latest version (which supports .NET Core 3.0)
- [x] Recompile QuickGraph and GraphX with support for .NET Standard or .NET Core 3.0
- [x] Migrate from `System.Windows.Interactivity` to `Microsoft.Xaml.Behaviors.Wpf`
- [ ] Decide how to get rid of Xceed.Datagrid
- [ ] Either wait until Xceed (other than Datagrid) release .NET Core version (in a few months?) or migrate to something else

**Comment by Kryptos-FR** [comment]
About QuickGraph, did you see that there is already a fork on GitHub that has still some development getting done?
https://github.com/YaccConstructor/QuickGraph

**Comment by Kryptos-FR** [comment]
@xen2 Makes sense. I was just wondering if forking their repo and starting from their initial releases (e.g. nuget https://www.nuget.org/packages/YC.QuickGraph/3.7.1) could work.

Might be safer to start from something closer to what we have currently though.

**Comment by meriaizen86** [comment]
Xceed now supports .Net Core 3.0.

https://xceed.com/release-notes/

**Comment by makotech222** [comment]
@meriaizen86 As discussed in the discord, they removed their datagrid component from the Free version, so we will have to replace the datagrid component with something else anyways.

**Comment by Kryptos-FR** [comment]
@meriaizen86 They might very well support .Net Core but it is still Windows-only. Xceed is and remains a WPF "extension", so it won't magically support other platforms.

**Comment by Jklawreszuk** [comment]
For those interested, it is possible to build mostly all Stride packages (except Tests) on Linux
result of 
`dotnet build Stride.sln`
![image](https://github.com/stride3d/stride/assets/31008367/088c8007-4fef-433c-aed2-e4639a466648)

**Comment by Jklawreszuk** [comment]
Since 4.2.0.2267 it is possible to build the project on Linux 💟 
Please keep in mind that It still may have some issues and support, so should be treated as in preview/beta.

If anyone is interested in supporting macOS you can follow my PR's for inspiration

**Comment by bruno-garcia** [comment]
I tried to build `dotnet build Stride.Runtime.sln` on a Mac and it fails with:

```
    stride/sources/targets/Stride.Core.targets(118,5): error MSB4062: The "AssemblyProcessorTask" task could not be loaded from the assembly /var/folders/3p/5s5zcvyn5491flwkcrqfh_880000gn/T/Stride/AssemblyProcessor/D51EE36025463AD7208310CB253CA306FB718893160EA88BD38C7203FCA2E0B7/netstandard2.0/Stride.Core.AssemblyProcessor.dll. Bad IL format. The format of the file '/var/folders/3p/5s5zcvyn5491flwkcrqfh_880000gn/T/Stride/AssemblyProcessor/D51EE36025463AD7208310CB253CA306FB718893160EA88BD38C7203FCA2E0B7/netstandard2.0/Stride.Core.AssemblyProcessor.dll' is invalid. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask.
```

**Comment by bruno-garcia** [comment]
After I installed `lld` on my Mac (`brew install lld`), I was able to run `dotnet build Stride.Runtime.sln`:

`Build succeeded with 1280 warning(s) in 21.3s`

**Comment by GothGirl-Gaming** [comment]
hi, i was hoping this issue was still up for grabs to debug. @stride3d/@stride-contributors
